source/particle/integrator/rk_2.cc

```diff
@@ -0,0 +1,141 @@
+/*
+  Copyright (C) 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/integrator/rk_2.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Integrator
+    {
+      template <int dim>
+      RK2<dim>::RK2()
+      {
+        step = 0;
+        loc0.clear();
+      }
+
+      template <int dim>
+      void
+      RK2<dim>::local_integrate_step(const typename std::multimap<types::LevelInd, Particle<dim> >::iterator &begin_particle,
+                                     const typename std::multimap<types::LevelInd, Particle<dim> >::iterator &end_particle,
+                                     const std::vector<Tensor<1,dim> > &old_velocities,
+                                     const std::vector<Tensor<1,dim> > &velocities,
+                                     const double dt)
+      {
+        typename std::multimap<types::LevelInd, Particle<dim> >::iterator it = begin_particle;
+        typename std::vector<Tensor<1,dim> >::const_iterator old_vel = old_velocities.begin();
+        typename std::vector<Tensor<1,dim> >::const_iterator vel = velocities.begin();
+
+        for (; it!=end_particle, vel!=velocities.end(), old_vel!=old_velocities.end(); ++it,++vel,++old_vel)
+          {
+            const types::particle_index id_num = it->second.get_id();
+            const Point<dim> loc = it->second.get_location();
+            if (step == 0)
```

Comment:
I think I would prefer if this variable's name would be changed from `step` to `sub_step` or even `integrator_substep`.